### PR TITLE
Add more asyncio able methods with unittests

### PR DESCRIPTION
**Pull request by swamper123**
Add method to be more DRY like
Add more asyncio able methods incl. unittests

Only problem is, that I can't verify them on a hardware, but the pattern of how these methods work seems to be the same.

### CODE REVIEW

File: snap7/client_async.py [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 import snap7
 from snap7.common import check_error
 from .client import Client
+from snap7.snap7types import buffer_type, buffer_size
+
 
 logger = logging.getLogger(__name__)
 
 
+def error_wrap(func):
```
Comment (gijzelaerr) [inline snap7/client_async.py:18]: why not import this from the other module?

File: snap7/client_async.py [hunk 2]
```diff
@@ -24,15 +36,15 @@ def __init__(self):
...
+        None - pass, like sync method without a receive check
         1 - Poll if result is available, otherwise do something else
         2 - wait_idle, do something else and then wait until an answer receives (not implemented, better use 1)
         3 - Callback a method/function if the answer receives (not implemented)
         :param mode: Mode how an async answer shall be handled
         :return:
         """
-        if mode not in [None, 1, 2]:
-            logger.warning(f"{mode} is not a legit mode. Has to be None, 1 or 2!")
+        if mode not in [None, 1, 2]:
```
Comment (gijzelaerr) [inline snap7/client_async.py:46]: useless end of line spaces

File: snap7/client_async.py (hunk 1 above)
Comment (swamper123) [reply snap7/client_async.py:18]: It is not imported in server.py, client.py and partner.py as well from anywhere.
Would util.py maybe a good place to store it there and then import it in these files?

File: snap7/client_async.py (hunk 2 above)
Comment (swamper123) [reply snap7/client_async.py:46]: Will be fixed in next push

File: snap7/client_async.py (hunk 1 above)
Comment (gijzelaerr) [reply snap7/client_async.py:18]: ah you are right, not sure why. Yeah, i think it could be a good one to centralize, but it is not crazy important.

Comment (swamper123) [reply snap7/client_async.py:18]: util.py seems to be for something else. I would leave it where it is at the moment and we should tidy it up in a later PR.

**Comment by swamper123** [comment]
Before this PR should be merged, #174  should be merged, so that the tests are correct.

**Pull request merged**